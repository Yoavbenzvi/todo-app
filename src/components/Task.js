import React from 'react';

const Task = ({ text, status, toggleTaskStatus, index, deleteTask }) => {
	const active = 'm-1 p-5 cursor-pointer border-l-8 border-orange-500 text-white flex justify-between'
	const disabled = 'm-1 p-5 cursor-pointer border-l-8 border-green-400 text-gray-500 flex justify-between'

	const deleteSelf = (e) => {
		deleteTask(index)
		e.stopPropagation();
	}

	return(
		<li 
			id='task'
			className={status ? active : disabled}
			onClick={() => toggleTaskStatus(index)}
		>
			<p 
				className={status ? '' : 'line-through'}
			>
				{text}
			</p> 
			<button 
				onClick={(e) => deleteSelf(e)} 
				className='text-red-500 text-xl font-bold no-underline'
			>
				X
			</button>
		</li>
	)
}

export default Task