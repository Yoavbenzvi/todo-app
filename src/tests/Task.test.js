import { shallow } from "enzyme";
import React from "react";
import Task from "../components/Task.js";

describe('Task component', () => {
	it("expect to render Task component", () => {
		
		const wrapper = shallow(<Task />)
		expect(wrapper).toMatchSnapshot();
	});

	it('test className', () => {
		const wrapper = shallow(<Task status={true}/>);
		const mockClass = 'm-1 p-5 cursor-pointer border-l-8 border-orange-500 text-white';

		expect(wrapper.find({id: 'task'}).hasClass(mockClass)).toEqual(true)
	})	
})

