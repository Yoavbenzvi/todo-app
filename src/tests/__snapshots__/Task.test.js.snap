// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component expect to render Task component 1`] = `
<li
  className="m-1 p-5 cursor-pointer border-l-8 border-green-400 text-white"
  id="task"
/>
`;
